include/SFML/Graphics/Rect.inl

```diff
@@ -106,27 +110,31 @@ bool Rect<T>::intersects(const Rect<T>& rectangle) const
 
 ////////////////////////////////////////////////////////////
 template <typename T>
-bool Rect<T>::intersects(const Rect<T>& rectangle, Rect<T>& intersection) const
+constexpr bool Rect<T>::intersects(const Rect<T>& rectangle, Rect<T>& intersection) const
 {
+    // Not using 'std::min' and 'std::max' to avoid depending on '<algorithm>'
+    const auto min = [](T a, T b){ return (a < b) ? a : b; };
+    const auto max = [](T a, T b){ return (a < b) ? b : a; };
```

Comment:
This is my deliberate style choice for writing lambdas. We don't have a style guide yet, so I don't really think it would be beneficial to find consensus on style during this PR. As a reminder, our priority regarding style should be to finalize #1861 rather than to perform small changes on a per-PR basis. 

Once we have `clang-format` in place, many of the decisions made here might not be relevant anymore. I don't think it's worthwhile use of our time to discuss minutiae like a single whitespace here, especially since we don't have existing practice to use as reference.

Reply:
Thought it might be a typo -- but fully agree, let's push those things to the formatting PR. 👍